lib/carelink/bolusNWizard.js

```diff
@@ -470,129 +472,164 @@ module.exports = function(timezone, model) {
   }
 
   var datumsBySeqNum = null, findNormal = null, findSquare = null, findWizard = null;
+  var pointer = 0, cachedObjects = null;
 
-  return function(simulator, datum, i, data) {
-    var normal = null, square = null, wizard = null;
-    var parsed = parser(datum);
+  function connectEvents(data) {
+    cachedObjects = [];
+    for (var i = 0; i < data.length; ++i) {
+      var normal = null, square = null, wizard = null;
+      var parsed = parser(data[i]);
 
-    if (datumsBySeqNum == null) {
-      buildCache(data);
-    }
+      if (datumsBySeqNum == null) {
+        buildCache(data);
+      }
 
-    if (findNormal == null) {
-      findNormal = makeFindNormalFn();
-    }
+      if (findNormal == null) {
+        findNormal = makeFindNormalFn();
+      }
 
-    if (findSquare == null) {
-      findSquare = makeFindSquareFn();
-    }
+      if (findSquare == null) {
+        findSquare = makeFindSquareFn();
+      }
 
-    if (findWizard == null) {
-      findWizard = makeFindWizardFn();
-    }
-    
-    if (parsed != null) {
-      switch(parsed.type) {
-        case 'bolusNormal':
-          if (isInCache(parsed)) {
-            removeFromCache(parsed);
-            normal = cleanEvent(parsed);
-            if (parsed.dualComponent) {
-              square = findSquare(parsed);
-              if (square != null) {
-                datumsBySeqNum[square.uploadId][square.uploadSeqNum] = null;
+      if (findWizard == null) {
+        findWizard = makeFindWizardFn();
+      }
+      
+      if (parsed != null) {
+        switch(parsed.type) {
+          case 'bolusNormal':
+            if (isInCache(parsed)) {
```

Comment:
Great, thanks. No need to change anything now, but in general, I like stuff like that to be in the comments. Code tells you what but not why -- so it's sometimes valuable to add some "why".